app/services/GraylogService.scala

```diff
@@ -0,0 +1,76 @@
+package services
+
+import java.io.InputStream
+import java.net.URLEncoder
+
+import jsonutils.JsonUtil
+import play.api.libs.json._
+import play.api.libs.json.Json._
+
+import play.api.Logger
+import play.api.libs.ws.WS.WSRequestHolder
+import play.api.libs.ws._
+
+import javax.inject.{Inject}
+import scala.collection.immutable.List
+import com.ning.http.client.Realm.AuthScheme
+
+import scala.collection.JavaConversions._
+import scala.concurrent.Await
+import scala.concurrent.duration.Duration
+
+/**
+ * Service to get extraction logs from Graylog.
+ */
+class GraylogService @Inject() extends LogService {
+  def getLog(extractorName: String, submissionID: Option[String]): List[String] = {
+    val playConfig = play.Play.application().configuration()
+    val serviceEndpoint = playConfig.getString("clowder.log.serviceEndpoint")
+    val username = playConfig.getString("clowder.log.username")
+    val password = playConfig.getString("clowder.log.password")
+    val prefix = playConfig.getString("clowder.log.extractorNamePrefix")
```

Comment:
I would suggest adding some error-handling here.
If any of these are empty, we should probably return / throw an error about the missing configuration.

Right now, I get an obscure error message about a bad URL scheme if the `clowder.log.serviceEndpoint` value isn't set:
`The URI scheme, of the URI null/api/search/universal/relative?query=application_name%3Anullncsa.image.preview*, must be equal (ignoring case) to 'http', 'https', 'ws', or 'wss'`

Also notice the `null` string in the middle of the resulting URL if `clowder.log.extractorNamePrefix ` isn't set.

Reply:
so, you define the variable as empty string or not defined in the conf file?

Reply:
do we allow clowder.log.extractorNamePrefix to be empty? or just check it is not defined.